File: src/snapshot.py
class Snapshot:
    def __init__(self):
        self.position = []
        self.snapshot_num = []

    def billboard_shot(self, movement):
        billboard_coordinate = (movement.x, movement.y)
        ''' if the coordinate is in the position list '''
        if billboard_coordinate in self.position:
            billboard_index = self.position.index(billboard_coordinate)
            self.snapshot_num[billboard_index] = \
                self.snapshot_num[billboard_index] + 1
        else:
            ''' if the coordinate is not in the position list
                insert the coordinate of the billboard to the list of position
            '''
            self.position.append(billboard_coordinate)
            self.snapshot_num.append(1)

    def unique_billboard_number(self):

        billboard_num = len(self.position)
        return billboard_num

File: src/main/scala/Snapshot.scala
import scala.collection.mutable.ListBuffer

class Snapshot {
  var position = new ListBuffer[(Int, Int)]
  var snapshotNum = new ListBuffer[Int]

  def billboardShot(movement: Movement) {
    val billboardCoordinate = (movement.x, movement.y)
    if (position contains billboardCoordinate) {
      val billboardIndex = position.indexOf(billboardCoordinate)
      snapshotNum(billboardIndex) = snapshotNum(billboardIndex) + 1
    } else {

      position += billboardCoordinate
      snapshotNum += 1
    }

  }

  def uniqueBillboardNum(): Int = {
    val billboardNum = position.length
    return billboardNum
  }
}
